### PR TITLE
fix: prevent crashes when unparsing count, extend, and append actions

### PR DESCRIPTION
**Pull request by arpitjain099**
**Type:  Bug**

## Description

Three of the action unparsers raise (or emit malformed output) on valid input that reaches them through the public `get_effective_command_line_invocation()`. I ran into these while reversing a parser that used `count`, `extend`, and `append` options, then reduced each to a minimal repro.

- `_unparse_count_action` builds the short flag with `flag[0] + flag[1] * count`. For a `count` option that was not given and whose default is `None`, the stored value is `None` and the multiplication raises `TypeError: can't multiply sequence by non-int of type 'NoneType'`. With a default of `0` (or a value equal to a nonzero default) `count` is `0`, so it emits a bare prefix like `-` instead of omitting the unused flag.
- `_unparse_extend_action` passes the stored values straight into `" ".join(...)`, so a typed option such as `type=int` raises `TypeError: sequence item 1: expected str instance, int found`. It also never calls `quote_arg_if_necessary`, unlike the `append` unparser, so a value containing spaces is emitted unquoted and the round trip breaks.
- `_unparse_append_action` indexes `values[0]` to detect the nested-list case, which raises `IndexError: list index out of range` when an `append` option with `default=[]` was not given (the `values is None` guard does not catch an empty list).

## Related Issues/PRs

None that I could find; I came across these while using the library.

## Motivation

`get_effective_command_line_invocation()` should either reproduce the invocation or omit an option that was not supplied. In these three cases it crashes or prints something that would not parse back, which defeats the point of the reconstructed command line.

## Implementation Details

Each fix is small and local to the action it concerns:

- `count`: return early when the value is `None`, and skip emitting anything when the effective count is `<= 0` (the flag was not given beyond its default).
- `extend`: quote each value with `quote_arg_if_necessary(str(value))`, matching what the `append` unparser already does. This fixes both the non-string crash and the unquoted-spaces round trip.
- `append`: return early when the normalized value list is empty, before the `values[0]` access.

I kept the changes minimal rather than refactoring the shared logic, but happy to consolidate if you would prefer.

## Testing

Added regression cases to the existing parametrized tests for all three actions (a not-given `count` with `None`/`0`/nonzero defaults, an `extend` with `type=int` and one with a space-containing value, and an `append` with `default=[]` that was not given). Confirmed each case failed before the change and passes after.

The full suite passes locally on Python 3.12:

```
75 passed in 0.07s
```

`ruff check`, `ruff format --check`, and `mypy` are also clean on both changed files.

## Documentation

No documentation changes; behavior of the public API is unchanged except that these inputs no longer crash or emit malformed output.

## Summary by Sourcery

Prevent crashes and malformed output when unparsing count, extend, and append actions from command-line invocations.

Bug Fixes:
- Skip emitting append options when the normalized value list is empty to avoid index errors.
- Avoid unparsing count options when the value is None or the effective count is non-positive, preventing type errors and stray flags.
- Quote and stringify extend action values before emitting them so non-string values and space-containing arguments are handled correctly.

Tests:
- Add parametrized regression tests covering edge cases for append, count, and extend actions, including absent options, defaults, and typed/space-containing values.

### CODE REVIEW

**Review by GhostofGoes — APPROVED** [review]

File: reverse_argparse/reverse_argparse.py [hunk 1]
```diff
@@ -400,6 +400,8 @@ def _unparse_append_action(self, action: Action) -> None:
         flag = self._get_option_string(action)
         if not isinstance(values, list):
             values = [values]
+        if not values:
+            return
```
[the remark this answers is not in the document]
Comment (jmgate) [reply reverse_argparse/reverse_argparse.py:404]: I've reviewed these changes, and they are indeed correct.  @sourcery-ai, are you saying we need to make similar changes to `_unparse_extend_action` as well?

**Comment by jmgate** [comment]
**praise:**  This is a fantastic contribution! 🎉 

Thanks so much, @arpitjain099, for finding these bugs, building out unit tests to capture them, and then offering up the solution.  Your thorough work is very much appreciated.  I'll see about getting this merged in soon.

File: reverse_argparse/reverse_argparse.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmgate) [reply reverse_argparse/reverse_argparse.py:404]: Filed #408 to look into this.
